reasons for different flux values:
- input parameters (obviously)
- float point rounding
- interpolation schemes
- grid resolution
- threading or cpu behavior

all can cascade into diferent values

expect idential I, Q, U, V to 5+ sigfigs

bitwise identical results

fluxes within 5-10%

similar polarization structure

consistent image morphology


timesteps can be generated from this formula: i.e. timestep 25000 = 5000 (in filename)

In [ ]:
import h5py, pathlib as p

for f in sorted(p.Path("/work/vmo703/ipole_outputs/Ma-0.5_5000")
                .glob("MAD_spin-0.5_t5000_RBETA_pos0.h5")):
    with h5py.File(f, "r") as H:
        tM = float(H["/fluid_header/t"][()]) # time in code units ~ M
        F  = float(H["/Ftot_unpol"][()])
    print(f.name, "  t[M] =", tM, "  Ftot[Jy] =", F)


MAD_spin-0.5_t5000_RBETA_pos0.h5   t[M] = 25000.006108902577   Ftot[Jy] = 0.5576462116678979


should produce output like:

MAD_spin-0.5_t5000_RBETA_pos0.h5   t[M] = 25000.006108902577   Ftot[Jy] = 0.5576462116678979

SANE+0.94_4000 (timestep: 20000) only has one value for MunitOffset and MunitSlope:

In [1]:
offset_slope_key = {
    "20000": {
        "MunitOffset": {
            "MAD+0.94": [
                4.5e+24, 2.5e+24, 3e+24, 2.2e+24
            ],
            "MAD-0.5": [
                1.3e+25, 8e+24, 1e+25, 8.5e+24
            ],
            "SANE-0.5": [
                5e+28,1.9e+28, 2.9e+28, 1.7e+28
            ],
            "SANE+0.94": [
                1.5e+27
            ]
        },
        "MunitSlope": {
            "MAD+0.94": [
                1.2, 1.2, 1.2, 1.1
            ],
            "MAD-0.5": [
                2.8, 2.9, 2.5, 2.1
            ],
            "SANE-0.5": [
                28.0, 30.0, 22.0, 21.0
            ],
            "SANE+0.94": [
                0.72
            ]
        }
    }
}

new PROBLEMS:

- using MunitUsed from IPOLE and using that for GRMONTY gives 0 flux - SANE only

conversion from nu*L_nu -> maybe find solid angle?

5.2.1 footnote 3 (Comparison of Sgr A* with GRMHD simulations) for conversion between I_nu and F_nu

set the munit in GRMONTY to match a data point near 230 GHz